code/modules/mining/machine_processing.dm

```diff
@@ -156,6 +169,89 @@
 	src.updateUsrDialog()
 	return
 
+/obj/machinery/mineral/processing_unit_console/proc/print_report(var/mob/living/user)
+	if(!inserted_id)
+		to_chat(user, span("warning", "No ID inserted. Cannot digitally sign."))
+		return
+	if(!input_mats.len && !output_mats.len && !alloy_mats)
+		to_chat(user, span("warning", "There is no data to print."))
+		return
+
+	var/obj/item/weapon/paper/P = new /obj/item/weapon/paper(loc)
+	var/date_string = worlddate2text()
+	idx++
+
+	var/form_title = "Form 0600 - Mining Yield Declaration"
+	var/dat = "<small><center><b>NanoTrasen Inc.<br>"
+	dat += "Civilian Branch of Operation</b><br><br>"
+
+	dat += "Form 0600<br> Mining Yield Declaration</center><hr>"
+	dat += "Facility: NSS Aurora<br>"
+	dat += "Date: [date_string]<br>"
+	dat += "Index: [idx]<br><br>"
+
+	dat += "Miner(s): <span class=\"paper_field\"></span><br>"
+	dat += "Ore Yields: <br>"
+
+	dat += "<table>"
+	for(var/material/OM in output_mats) {
+		if(output_mats[OM] > 1)
+			dat += "<tr><td><b><small>[output_mats[OM]]</b></small></td><td><small>[OM.display_name] [OM.sheet_plural_name]</small></td></tr>"
+		else
+			dat += "<tr><td><b><small>[output_mats[OM]]</b></small></td><td><small>[OM.display_name] [OM.sheet_singular_name]</small></td></tr>"
+	}
+	for(var/datum/alloy/AM in alloy_mats) {
+		if(alloy_mats[AM] > 1)
+			dat += "<tr><td><b><small>[alloy_mats[AM]]</b></td><td><small>[AM.metaltag] sheets</small></td></tr>"
+		else
+			dat += "<tr><td><b><small>[alloy_mats[AM]]</b></td><td><small>[AM.metaltag] sheet</small></td></tr>"
+	}
+
+	dat += "</table><br>"
+
+	if(waste > 0)
+		dat += "Waste detected: "
+		dat += "[waste] unit(s) of material were wasted due to operator error. This includes: <br>"
+		dat += "<table>"
+		for(var/ore/IM in input_mats) {
+			if(input_mats[IM] > 1)
+				dat += "<tr><td><b><small>[input_mats[IM]]</small></b></td><td><small>[IM.display_name]</small></td></tr>"
+			else
+				dat += "<tr><td><b><small>[input_mats[IM]]</small></b></td><td><small>[IM.display_name]</small></td></tr>"
+		}
+		dat += "</table><br>"
+
+	dat += "Additional Notes: <span class=\"paper_field\"></span><br><br>"
+
+	dat += "Quartermaster's / Head of Personnel's / Captain's Stamp: </small>"
+
+	P.set_content(form_title, dat)
+
+	//stamp the paper
+	var/image/stampoverlay = image('icons/obj/bureaucracy.dmi')
+	stampoverlay.icon_state = "paper_stamp-cent"
+	if(!P.stamped)
+		P.stamped = new
+	P.offset_x += 0
+	P.offset_y += 0
+	P.ico += "paper_stamp-cent"
+	P.stamped += /obj/item/weapon/stamp
+	P.add_overlay(stampoverlay)
+	P.stamps += "<HR><i>This paper has been stamped by the NT Ore Processing System.</i>"
+
+	playsound(loc, "sound/bureaucracy/print.ogg", 75, 1)
+	user.visible_message("\The [src] spits out a piece of paper.")
+
+	// reset
+	output_mats = list()
+	input_mats = list()
+	waste = 0
+
+	if(istype(user,/mob/living/carbon/human) && !(user.l_hand && user.r_hand))
```

Comment:
Has this been tested on cyborgs? Because they should be able to use this.

Reply:
Yes, that check is purely to put the paper in a person's hands; a cyborg using the machine will have the paper placed on their turf (in the latest commit, anyway)